## **Problem and Algortithm**

Terdapat food delivery dataset yang akan difilter pesanan yang berhasil dipesan dan yang tidak jadi dipesan. Setelah itu akan dicari harga makanan termurah customer dan harga makanan termahal customer, maka algoritma untuk menentukannya yaitu:
* Connect GDrive
* Import Library dan Dataset
* Memfilter kolom status yang "Completed"
* Memfilter kolom status yang "Cancelled"
* Mengurutkan kolom food_price dari yang terkecil
* Mengurutkan kolom food_price dari yang terbesar

**1. Connect GDrive**

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


**2. Import Library dan Data**

In [20]:
import pandas as pd
dataset = pd.read_csv('sample_data/food_delivery_datasets.csv')
dataset

order_id  voucher_amount  ...     status  random_id
0     FOOD-10455               0  ...  Completed   0.511149
1     FOOD-10985               0  ...  Completed   0.832936
2     FOOD-10911               0  ...  Completed   0.828026
3     FOOD-10932               0  ...  Completed   0.239377
4     FOOD-10961               0  ...  Completed   0.995587
...          ...             ...  ...        ...        ...
1071  FOOD-10921               0  ...  Completed   0.248328
1072  FOOD-10265            5000  ...  Completed   0.268251
1073  FOOD-10534               0  ...  Completed   0.132528
1074  FOOD-10158               0  ...  Completed   0.167946
1075  FOOD-10770               0  ...  Completed   0.358051

[1076 rows x 11 columns]

Terlihat ada 1076 data pesanan

**3. Memfilter kolom status yang "Completed"**

In [25]:
dataset[dataset["status"]=="Completed"]

order_id  voucher_amount  ...     status  random_id
0     FOOD-10455               0  ...  Completed   0.511149
1     FOOD-10985               0  ...  Completed   0.832936
2     FOOD-10911               0  ...  Completed   0.828026
3     FOOD-10932               0  ...  Completed   0.239377
4     FOOD-10961               0  ...  Completed   0.995587
...          ...             ...  ...        ...        ...
1071  FOOD-10921               0  ...  Completed   0.248328
1072  FOOD-10265            5000  ...  Completed   0.268251
1073  FOOD-10534               0  ...  Completed   0.132528
1074  FOOD-10158               0  ...  Completed   0.167946
1075  FOOD-10770               0  ...  Completed   0.358051

[881 rows x 11 columns]

Terlihat ada 881 data yang menyatakan status pesanan "Completed" 

**4. Memfilter kolom status yang "Cancelled"**

In [26]:
dataset[dataset["status"]=="Cancelled"]

order_id  voucher_amount  ...     status  random_id
20    FOOD-10888               0  ...  Cancelled   0.208880
22    FOOD-10394               0  ...  Cancelled   0.640676
29    FOOD-10731               0  ...  Cancelled   0.233675
30    FOOD-10230               0  ...  Cancelled   0.743930
34    FOOD-10584               0  ...  Cancelled   0.840597
...          ...             ...  ...        ...        ...
1028  FOOD-10839               0  ...  Cancelled   0.943491
1043  FOOD-10229               0  ...  Cancelled   0.483035
1057  FOOD-10957               0  ...  Cancelled   0.726522
1059  FOOD-10119               0  ...  Cancelled   0.580990
1069  FOOD-10459               0  ...  Cancelled   0.668113

[195 rows x 11 columns]

Terlihat ada 195 data yang menyatakan status pesanan "Cancelled" 

**5. Mengurutkan kolom food_price dari yang terkecil**

In [24]:
df.sort_values('food_price')

order_id  voucher_amount  ...     status  random_id
202   FOOD-10654               0  ...  Completed   0.716985
412   FOOD-10898               0  ...  Completed   0.751653
858   FOOD-10395               0  ...  Completed   0.093072
851   FOOD-11013               0  ...  Completed   0.753612
864   FOOD-10439               0  ...  Completed   0.815192
...          ...             ...  ...        ...        ...
1037  FOOD-10781               0  ...  Completed   0.674128
827   FOOD-10565               0  ...  Completed   0.912690
67    FOOD-10633               0  ...  Completed   0.734508
10    FOOD-10386               0  ...  Completed   0.485939
128   FOOD-10212               0  ...  Completed   0.469235

[1076 rows x 11 columns]

Terlihat bahwa harga makanan termurah yang dipesan oleh customer yaitu Rp 12.900,00

**6. Mengurutkan kolom food_price dari yang terbesar**

In [ ]:
df.sort_values('food_price',ascending=False)

order_id  voucher_amount  ...     status  random_id
128   FOOD-10212               0  ...  Completed   0.469235
10    FOOD-10386               0  ...  Completed   0.485939
67    FOOD-10633               0  ...  Completed   0.734508
827   FOOD-10565               0  ...  Completed   0.912690
1037  FOOD-10781               0  ...  Completed   0.674128
...          ...             ...  ...        ...        ...
864   FOOD-10439               0  ...  Completed   0.815192
851   FOOD-11013               0  ...  Completed   0.753612
858   FOOD-10395               0  ...  Completed   0.093072
412   FOOD-10898               0  ...  Completed   0.751653
202   FOOD-10654               0  ...  Completed   0.716985

[881 rows x 11 columns]

Terlihat bahwa harga makanan termahal yang dipesan oleh customer yaitu Rp 448.500,00